### DISCUSSION TITLE
How to debug?

### DISCUSSION THREAD

**Discussion by cederron** [prevwong/craft.js#402] [Q&A]
Sorry for this basic question, i feel it's more of a webpack/vs code/node question, but since i want to work with craftjs i thought you might be willing to help me.

I'm using vs code.
I followed doc steps , basically get source and "yarn install/yarn dev".  The app launches.
On this landing page i can see source code and setting breakpoints work as expected, but if i try to debug code from craftjs (some source code in packages/core/* for example)  it's not working as expected.

If I inspect from chrome dev tools all I can find is semi-compiled code from /dist folder, if i set a breakpoint there, vs-code shows up but using this same pre-compiled code file. 
If I set breakpoints in vscode in source file they are not triggered (I have attached vscode debugger prior to this, maybe i attached it wrong...?).
So at the moment i dont' know how to set/trigger breakpoints using source code.

Any help would be appreciated. Thanks.

**Comment by prevwong** [comment]
Hey there, I don't really have experience with using breakpoints in VSCode. But I wonder if generating sourcemaps could help solve this issue?  So maybe setting `sourcemap: true` in the `output` items here: https://github.com/prevwong/craft.js/blob/develop/rollup.config.js 

Could you try doing that and see if that works?

**Reply by cederron** [reply]
> Hey there, I don't really have experience with using breakpoints in VSCode. But I wonder if generating sourcemaps could help solve this issue? So maybe setting `sourcemap: true` in the `output` items here: https://github.com/prevwong/craft.js/blob/develop/rollup.config.js
> 
> Could you try doing that and see if that works?

I tried , but it didn't work.
What do you use to develop/debug craftjs ?
